swig/python/test_comedi_command.py

```diff
@@ -0,0 +1,161 @@
+
+        nchans = len(chans)  # number of channels
+
+        # wrappers include a "chanlist" object (just an Unsigned Int
+        # array) for holding the chanlist information
+        mylist = c.chanlist(nchans)  # create a chanlist of length nchans
+
+        # now pack the channel, gain and reference information into
+        # the chanlist object
+        # N.B. the CR_PACK and other comedi macros are now python functions
+        for index in range(nchans):
+            mylist[index] = c.cr_pack(
+                chans[index], gains[index], aref[index])
+
+        # construct a comedi command manually
+        # TODO: make helper methods
+        cmd = c.comedi_cmd_struct()
+        cmd.subdev = 0
+        cmd.flags = 0
+        cmd.start_src = c.TRIG_NOW
+        cmd.sart_arg = 0
+        cmd.scan_begin_src = c.TRIG_TIMER
+        cmd.scan_begin_arg = int(1.0E9 / 100000)
+        cmd.convert_src = c.TRIG_TIMER
+        cmd.convert_arg = 1
+        cmd.scan_end_src = c.TRIG_COUNT
+        cmd.scan_end_arg = nchans
+        cmd.stop_src = c.TRIG_COUNT
+        cmd.stop_arg = nscans
+        cmd.chanlist = mylist
+        cmd.chanlist_len = nchans
+
+        dev = self.dev
+        # test our comedi command a few times.
+        ret = c.comedi_command_test(dev, cmd)
+        print("first cmd test returns ", ret)
+        ret = c.comedi_command_test(dev, cmd)
+        print("second test returns ", ret)
+        ret = c.comedi_command_test(dev, cmd)
+        if not ret:
+            raise "Error testing comedi command"
+        self.cmd = cmd
+
+        ret = c.comedi_command(self.dev, self.cmd)
+        return ret
+
+    def readcommand(self):
+        """Read some data from a running command
+        [sic] I think this will work but it's completely untested! --bryan
+        """
+        datalist = []
+        data = os.read(self.cfd)
+        while len(data) > 0:
+            datalist.append(data)
+            data = os.read(self.cfd)
+
+        datastr = string.join(datalist, '')
+
+        print(datastr)  # Here's your data, as a single string!
+        # if you've got Numeric installed you can convert data
+        # into a flat Numpy array with:
+        # dataarray = Numeric.fromstring(data, Int16)
+
+    def close(self):
+        'close and cleanup'
+        ret = c.comedi_close(self.dev)
+        return ret
+
+if __name__ == "__main__":
+    BASEDIR = os.path.abspath(os.path.dirname(sys.argv[0]))
+
+    PARSER = argparse.ArgumentParser(
+        description='comedilib python SWIG command test',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    # puts the defaults in the help automatically!
+    PARSER.add_argument('-d', '--debug', action='store_true',
+                        help='Set debug flags')
+    PARSER.add_argument('-i', '--input', default='/dev/comedi0',
+                        help='comedi device, default /dev/comedi0')
+
+    ARGS = PARSER.parse_args()
+    CET = COMEDILIBtest(ARGS.input)
```

Comment:
This is bit of nitpicking, but usually in Python, only constants are all upper-case. I'd recommend to use lowercase for parser, args, and cet.